Update and Improve install.md

klees opened the pull request:
Hi folks,

I had an intense look into the install.md and this is what I came up with. I restructured a lot of stuff, removed some outdated or duplicate information and incorporated information about the command line setup. This also includes the changes from #2978 and the problems adressed in #3082, at least I tried. I also removed the RFC-2119 language, because this is no RFC but a howto.

I would appreciate if someone other then me could have a long look into this as well.

Best regards!

klees committed:
install.md: use intro from ilias.de

klees committed:
install.md: huge cleanup and reorganisation

klees committed:
install.md: document cli-setup

Amstutz reviewed (approved):
Hi @rklees

Thx very much for this much needed work-over of the install docs. Much appreciated. I am no sysadmin, so I can not give much feedback on large parts of the content. Hover, I like the general direction towards which you pulled the install docs. IMO they seem a bit more manageable this way.

Consider the following suggestions:
- [x] Release numbers: Refrain from using exact release numbers anywhere. They tend to get outdated. Instead of release_7, use release_X (maybe with the comment the X should be replaced for the targeted number.
- [x] System Styles: On old line 573 you removed an old pointer to some section which was not helpful in terms of system styles. Thx. However, it would be neat to replace it with the really helpful docs at: [Custom
Styles](/templates/Readme.md#custom-styles) . They tend to be overlooked. So an additional pointer here would be neat.
- [x] Current DB step: I personally found the tip concerning checking up on the current DB step really neat:To check which update step gets currently executed run the following SQL-Statement on your ILIAS database: `SELECT * FROM settings WHERE keyword = "db_update_running"`. As long as the script is not spitting out those numbers, I would keep it here. Can really save the day.

Take or leave, but mainly thx very very much.

kergomard commented on docs/configuration/install.md:
```diff
@@ -1,39 +1,47 @@
 # ILIAS Installation
 
-ILIAS is a powerful Open Source Learning Management System for developing and realising web-based e-learning. The software was developed to reduce the costs of using new media in education and further training and to ensure the maximum level of customer influence in the implementation of the software. ILIAS is published by ILIAS open source e-Learning e.V. under the General Public Licence and fre ...
+Around since 1998, ILIAS is a powerful learning management system that fulfils
```
```suggestion
Since around 1998, ILIAS is a powerful learning management system that fulfils
```

klees replied:
I'll simply remove this, probably does not belong in installation instructions anyway.

kergomard commented on docs/configuration/install.md:
```diff
@@ -56,141 +62,99 @@ ILIAS is a powerful Open Source Learning Management System for developing and re
-## Bandwidth
-
-We RECOMMEND at least 100 Mbit/sec. for the web server WAN connection.
+We recommend a **common dual core server CPU**. Memory requirements are greatly
```
Do we really want this block? Personally I would never run ILIAS on a setup like this unless I'm running a very small instance with all goodies switched off. Our instance is not huge, but I'm still counting around 50 Cores, multiple Terabytes of data (without counting RAIDs and failovers) and 100's of MB of RAM. Additionally running ILIAS on a hard drive will be painful and clearly not comply with today's demands for snappiness. 100 Mbits does also sound slightly 1990's ;-).

Just a suggestion from where I stand we can also simply throw it out:
"The hardware requirements for ILIAS vary widely, depending on the number of concurrent users you expect and the features you want to enable. Please be aware that ILIAS is not a webpage, but a highly interactive application, thus requirements will be higher than in case of the former. Snappiness of the system will highly depend on deploying enough resources and tailoring the system to your needs. In any case we recommend an absolute minimum of a **common dual core server CPU**, **4GB of RAM**, and a **100 Mbit/s** internet connection. Roughly estimated the operating system and ILIAS itself will use around 25GB. From there you can calculate your storage needs based on the amount of files and media content you expect to upload plus a few GBs for the database.

kergomard commented on docs/configuration/install.md:
````diff
@@ -56,141 +62,99 @@ ILIAS is a powerful Open Source Learning Management System for developing and re
-composer install --no-dev
-```
+Please note that other platforms and configurations should be possible, but it
+might be harder to find someone who can help when things go south. You shall not
````
Sounds a little bit like Gandalf :-)
```suggestion
might be harder to find someone who can help when things go south. You should not
```

kergomard commented on docs/configuration/install.md:
````diff
@@ -249,31 +214,34 @@ On RHEL/CentOS:
+On Debian/Ubuntu execute:
 ```
-apt-get install libapache2-mod-php7.1 php7.1-gd php7.1-mysql php7.1-mbstring php7.1-curl php7.1-dom php7.1-zip php-xml
+apt-get install libapache2-mod-php7.3 php7.3-gd php7.3-mysql php7.3-mbstring php7.3-curl php7.3-dom php7.3-zip php7.3-xml
````
I would recommend to get rid of this. It is hard to keep track of and changes can be sudden. Additionally, if you need this instructions and follow them by the letter, you may end up with a system including eol packages really quickly.

My suggestion:
```suggestion
Please refere to the documentation of your installation to install either php7.3 or php7.4 including the gd, mysql, mbstring, curl, dom, zip, and xml modules. 
```

kergomard commented on docs/configuration/install.md:
```diff
@@ -502,93 +541,223 @@ ExecStop=/usr/bin/java $JAVA_OPTS -jar $ILSERVER_JAR $ILSERVER_INI stop
+an empty files, which results in an empty file object on ILIAS. Due to a bug in
+apache, the configuration of **Apache with PHP-FPM** does not work with the *Mac
+Finder*. If you use WebDAV on your ILIAS installation, we recommend to either
+use **Apache with mod_php** or **Nginx with PHP-FPM (> 1.3.8)**.
```
```suggestion
use **Apache with mod_php** or **Nginx with PHP-FPM (> 1.3.8)**. Please find more information in the corresponding [README](../../Services/WebDAV/README.md)
```
...and then we get rid of the rest of the information in here concerning WebDAV. We might want to do this for all optional elements to keep this file short and hand responsibility over to maintainers. What do you think?

klees replied:
Something like this would be great.

kergomard commented on docs/configuration/install.md:
````diff
@@ -361,108 +333,169 @@ mysql -i -BN -e 'SELECT @@sql_mode' | grep -E 'ONLY_FULL_GROUP_BY|STRICT_TRANS_T
+chown www-data:www-data `/var/www/files
+```
+
+If you want to server other applications from your webserver, you might want to
````
```suggestion
If you want to serve other applications from your webserver, you might want to
```
...but I would actually suggest to kick that part out and only keep "The commands above will directly serve ILIAS from the docroot." I don't see many good reasons to want to serve ILIAS from subdirectories. Certificates are free (or at least getting cheaper if you want e.Vs.) and subdomains seem by far more elegant to me.

kergomard commented on docs/configuration/install.md:
```diff
@@ -361,108 +333,169 @@ mysql -i -BN -e 'SELECT @@sql_mode' | grep -E 'ONLY_FULL_GROUP_BY|STRICT_TRANS_T
-RewriteCond %{REQUEST_METHOD} ^(OPTIONS)$
-RewriteRule .* "-" [R=401,L]
+cd /var/www/html/
+git clone https://github.com/ILIAS-eLearning/ILIAS.git .
```
We could suggest to use --single-branch here. For me the difference when testing was 500MB of dead waste.

kergomard reviewed (commented):
Thank you very much @klees for all the work! I added comments inline, as this is not really code, but if you prefer them in our standardized form here at the end. I will do so, just let me know.

kergomard commented on docs/configuration/install.md:
````diff
@@ -361,108 +333,169 @@ mysql -i -BN -e 'SELECT @@sql_mode' | grep -E 'ONLY_FULL_GROUP_BY|STRICT_TRANS_T
+Run the ILIAS command line setup with your configuration file from your ILIAS folder:
+
+```
+php setup/setup.php install my-configuration.json
````
```suggestion
php setup/setup.php install /foo/bar/my-configuration.json
```

kergomard commented on docs/configuration/install.md:
````diff
@@ -361,108 +333,169 @@ mysql -i -BN -e 'SELECT @@sql_mode' | grep -E 'ONLY_FULL_GROUP_BY|STRICT_TRANS_T
+}
+```
+
+Run the ILIAS command line setup with your configuration file from your ILIAS folder:
````
```suggestion
Run the ILIAS command line setup from within your ILIAS folder passing your configuration file as a parameter:
```

kergomard reviewed (commented):
Making it clearer that the config.json has to be outside the webroot. I landed on /foo/bar as a folder, as I was unsure what else to pick: /opt/, /etc/ilias/, /home/ilias ?

klees committed:
Docs: removed intro in install.md

This simply does not belong there and would require additional maintenance effort
that we could better use to do something else.

klees committed:
Docs: improve install.md (thx @swiniker and @Amstutz)

klees commented:
Hi @Amstutz,

the doc for the custom styles is (and was) linked here:

https://github.com/ILIAS-eLearning/ILIAS/pull/3158/files#diff-703d97b1d89d3339a2aab21d2d33190b3b9fa8b044efd785a17ff91ba1cc63e6R594

Was it something else you meant?

Best regards!

Amstutz commented:
You are right concerning the custom styles, got confused with the deleted headline above ;-).

I believe this is an important step. It is sad that there has not been more feedback on this important work. However, I guess that silence means approval. I would support merging this important contribution.

Thx a lot for doing that.

klees commented:
Hi Jour Fixe,

there is nothing to decide for you here. I still want to bring this to the attention of the JF: We have a problem with our documentation. Nobody seems to really feel responsible for such an important document like the installation instructions. We need to do better here.

Best regards!

matthiaskunkel commented:
Jour Fixe, 08 MAR 2021: Thanks for pointing us at this issue. A feasible and sustainable solution for maintaining the installation documentation is highly appreciated and important to extend the user base of ILIAS. We are looking forward to the suggestions made by the TB at the upcoming DevConf.

klees commented:
I have picked this to trunk.
